File: testers/avro/scala-json/generated-and-checked-in/com/example/events/StringOrLong.scala
package com.example.events

import java.lang.UnsupportedOperationException

/** Union type for: string | long */
sealed trait StringOrLong {
  /** Check if this union contains a string value */
  def isString: Boolean

  /** Get the string value. Throws if this is not a string. */
  def asString: String

  /** Check if this union contains a long value */
  def isLong: Boolean

  /** Get the long value. Throws if this is not a long. */
  def asLong: Long
}

object StringOrLong {
  /** Create a union value from a string */
  def of(value: String): StringOrLong = {
    return new StringValue(value)
  }

  /** Create a union value from a long */
  def of(value: Long): StringOrLong = {
    return new LongValue(value)
  }

  /** Wrapper for long value in union */
  case class LongValue(value: Long) extends StringOrLong {
    override def isString: Boolean = {
      return false
    }

    override def asString: String = {
      throw new UnsupportedOperationException("Not a String value")
    }

    override def isLong: Boolean = {
      return true
    }

    override def asLong: Long = {
      return value
    }
  }

  /** Wrapper for string value in union */
  case class StringValue(value: String) extends StringOrLong {
    override def isString: Boolean = {
      return true
    }

    override def asString: String = {
      return value
    }

    override def isLong: Boolean = {
      return false
    }

    override def asLong: Long = {
      throw new UnsupportedOperationException("Not a Long value")
    }
  }
}

File: testers/avro/java-vanilla/generated-and-checked-in/com/example/events/StringOrLong.java
package com.example.events;

/** Union type for: string | long */
public sealed interface StringOrLong permits StringOrLong.StringValue, StringOrLong.LongValue {
  /** Wrapper for long value in union */
  record LongValue(Long value) implements StringOrLong {
    public LongValue withValue(Long value) {
      return new LongValue(value);
    }

    @Override
    public java.lang.String toString() {
      return value.toString();
    }

    @Override
    public Long asLong() {
      return value;
    }

    @Override
    public String asString() {
      throw new UnsupportedOperationException("Not a String value");
    }

    @Override
    public Boolean isLong() {
      return true;
    }

    @Override
    public Boolean isString() {
      return false;
    }
  }

  /** Wrapper for string value in union */
  record StringValue(String value) implements StringOrLong {
    public StringValue withValue(String value) {
      return new StringValue(value);
    }

    @Override
    public java.lang.String toString() {
      return value.toString();
    }

    @Override
    public Long asLong() {
      throw new UnsupportedOperationException("Not a Long value");
    }

    @Override
    public String asString() {
      return value;
    }

    @Override
    public Boolean isLong() {
      return false;
    }

    @Override
    public Boolean isString() {
      return true;
    }
  }

  /** Create a union value from a string */
  static StringOrLong of(String value) {
    return new StringValue(value);
  }

  /** Create a union value from a long */
  static StringOrLong of(Long value) {
    return new LongValue(value);
  }

  /** Get the long value. Throws if this is not a long. */
  Long asLong();

  /** Get the string value. Throws if this is not a string. */
  String asString();

  /** Check if this union contains a long value */
  Boolean isLong();

  /** Check if this union contains a string value */
  Boolean isString();
}
